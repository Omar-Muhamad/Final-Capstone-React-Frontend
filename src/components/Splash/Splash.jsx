import { NavLink } from 'react-router-dom';
import logo from '../../img/microvago-logo-i.png';

const Splash = () => (
  <div className="splashBG h-screen w-full flex flex-col  items-center justify-center bg-[#6D22FB]">
    <div className="grow-[4] sm:w-[40%] w-[80%] flex items-center justify-center">
      <img src={logo} alt="Logo of MicroVago" />
    </div>
    <div className="AuthenticationBtns grow flex flex-col gap-4 text-xl font-medium text-center w-full">
      <NavLink
        to="/login"
        className="w-[70%] lg:w-1/4 mx-auto py-4 rounded-full mx-auto bg-white text-[#6D22FB] text-white cursor-pointer"
      >
        Login
      </NavLink>
      <NavLink to="/signup" className="w-[70%] lg:w-1/4 mx-auto py-4 border-2 border-white text-white rounded-full cursor-pointer">
        Sign up
      </NavLink>
    </div>
  </div>
);

export default Splash;
