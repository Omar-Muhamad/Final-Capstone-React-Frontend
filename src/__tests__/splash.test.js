import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import Splash from '../components/Splash/Splash';
import store from '../redux/configureStore';

describe('Components testing', () => {
  test('Splash component', () => {
    const tree = render(<Provider store={store}><BrowserRouter><Routes><Route path="/" element={<Splash />} /></Routes></BrowserRouter></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
