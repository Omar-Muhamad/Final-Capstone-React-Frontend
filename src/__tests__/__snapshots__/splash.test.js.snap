// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components testing Splash component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="splashBG h-screen w-full flex flex-col  items-center justify-center bg-[#6D22FB]"
      >
        <div
          class="grow-[4] w-1/3 flex items-center justify-center"
        >
          <img
            alt="Logo of MicroVago"
            src="logo.png"
          />
        </div>
        <div
          class="AuthenticationBtns grow flex flex-col gap-4 text-xl font-medium text-center w-full"
        >
          <a
            class="w-[70%] lg:w-1/4 mx-auto py-4 rounded-full mx-auto bg-white text-[#6D22FB] text-white cursor-pointer"
            href="/login"
          >
            Login
          </a>
          <a
            class="w-[70%] lg:w-1/4 mx-auto py-4 border-2 border-white text-white rounded-full cursor-pointer"
            href="/signup"
          >
            Sign up
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="splashBG h-screen w-full flex flex-col  items-center justify-center bg-[#6D22FB]"
    >
      <div
        class="grow-[4] w-1/3 flex items-center justify-center"
      >
        <img
          alt="Logo of MicroVago"
          src="logo.png"
        />
      </div>
      <div
        class="AuthenticationBtns grow flex flex-col gap-4 text-xl font-medium text-center w-full"
      >
        <a
          class="w-[70%] lg:w-1/4 mx-auto py-4 rounded-full mx-auto bg-white text-[#6D22FB] text-white cursor-pointer"
          href="/login"
        >
          Login
        </a>
        <a
          class="w-[70%] lg:w-1/4 mx-auto py-4 border-2 border-white text-white rounded-full cursor-pointer"
          href="/signup"
        >
          Sign up
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
